### ISSUE TITLE
error on remove_control_from_docks() if EditorPlugin script altered

### ISSUE THREAD

**Issue by hedgehog90** [godotengine/godot#19085]
```
tool
extends EditorPlugin

var dock

func _enter_tree():
	dock = preload("Panel.tscn").instance()
	add_control_to_dock(DOCK_SLOT_LEFT_BR, dock)

func _exit_tree():
	remove_control_from_docks(dock) # Remove the dock
	dock.free() # Erase the control from the memory
```

If the script is altered in any way (even adding a return), on _exit_tree(), dock is null:

```
 editor\editor_plugin.cpp:325 - Parameter ' p_control ' is null.
 res://addons/Tools/tools.gd:25 - Invalid call. Nonexistent function 'free' in base 'Nil'.
```

**Comment by Gibbz** [comment]
Yes i keep bumping into this bug! Really annoying when trying to work on a plugin. It keeps spawning a new dock each time and never removes the old

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

If yes, please ensure that an up-to-date Minimal Reproduction Project (MRP) is included in this report (a MRP is a zipped Godot project with the minimal elements necessary to reliably trigger the bug). You can upload ZIP files in an issue comment with a drag and drop.

**Comment by akien-mga** [comment]
No answer, so closing.

**Comment by KoBeWi** [comment]
I actually encountered this problem recently, but there's a proposal that covers this issue: https://github.com/godotengine/godot-proposals/issues/1012

**Comment by leetNightshade** [comment]
I see this issue on 3.3.3 stable.

**Comment by Patchcoat** [comment]
This post still shows up both in google and this problem still shows up in 3.5, so for future reference I was able to remove the dock using queue_free() instead of free()

**Comment by KoBeWi** [comment]
Eh it doesn't matter if you use `free()` or `queue_free()` here. The problem is that the instance was being lost on script save, so it couldn't be removed. This is fixed in 4.0.

Also the reason why it worked in your case is probably that you used `free()` before `remove_control_from_docks()` (you should do it after remove).

**Comment by Patchcoat** [comment]
I'm using it after `remove_control_from_docks()`, so I tested it again and it's back to not working. The reason it was working was probably because I had restarted between changing between free to queue_free or something, so the script didn't have a chance to loose it's reference from the time it called `_enter_tree()` to the time I toggled the plugin.

Good to know it won't be a problem when I distribute it, and it won't be a problem _at all_ after my team migrates to 4.0 sometime next year.
